arches_lingo/src/arches_lingo/components/generic/ResourceInstanceRelationships.vue

```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+import ResourceInstanceRelationshipsViewer from "@/arches_lingo/components/generic/ResourceInstanceRelationshipsViewer.vue";
+import ResourceInstanceRelationshipsEditor from "@/arches_lingo/components/generic/ResourceInstanceRelationshipsEditor.vue";
+import { EDIT, VIEW } from "@/arches_lingo/constants.ts";
+import type {
+    DataComponentMode,
+    ResourceInstanceReference,
+} from "@/arches_lingo/types";
+
+const { mode = EDIT } = defineProps<{
+    mode?: DataComponentMode;
+    value?: ResourceInstanceReference[];
+    options?: ResourceInstanceReference[];
+}>();
+
+const emits = defineEmits(["update"]);
+
+function onUpdate(val: string[]) {
+    emits("update", val);
+}
+</script>
+<template>
+    <div v-if="mode === VIEW">
+        <ResourceInstanceRelationshipsViewer :value="value" />
+    </div>
+    <div v-if="mode === EDIT">
+        <ResourceInstanceRelationshipsEditor
+            :options="options"
+            :val="value?.map((x) => x.resourceId)"
```

Comment:
ts error goes away with:
```suggestion
            :val="value?.map((x) => x.resourceId) ?? []"
```

Reply:
never mind, not seeing this anymore